src/Iter.mo

```diff
@@ -32,6 +31,19 @@ module {
     }
   };
 
+  public func singleton<T>(value : T) : Iter<T> {
+    object {
+      var returned = false;
+      public func next() : ?T {
+        if returned { null }
+        else {
+          returned := true;
+          ?value
+        }
+      }
+    }
+  };
```

Comment:
```suggestion
    object {
      var state= ?value;
      public func next() : ?T {
        switch state { 
           case null null;
           case some {
             state := null;
             some
           }
        }
      }
    }
  };
```

Might be little more efficient and avoid a small space leak (the closure for next holds on to value above, even once consumed).

Reply:
I'll open this in a separate PR as a workaround for needing a reapproval.